core/ir/synthetic/encoding_common.h

```diff
@@ -0,0 +1,147 @@
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef _SYNTHETIC_ENCODING_COMMON_H_
+#define _SYNTHETIC_ENCODING_COMMON_H_
+
+/**
+ * This synthetic ISA (which we call DR_ISA_REGDEPS) is a made up ISA that has the purpose
+ * of preserving register dependencies and giving hints on the type of operations each
+ * instruction is performing.
+ * For this reason the majority of operations that would normally work on instructions
+ * coming from an actual ISA (e.g., DR_ISA_AMD64) are not supported.
+ * We support instr_convert_to_isa_regdeps(), which converts an #instr_t of an actual ISA
+ * to a DR_ISA_REGDEPS instruction.
+ * Other operations we support (related to encoding and decoding of a DR_ISA_REGDEPS
+ * #instr_t) are: instr_encode(), instr_encode_to_copy(), decode(), decode_common().
+ * DR_ISA_REGDEPS #instr_t can only return the encoded information described in the
+ * encoding scheme (core/ir/synthetic/encoding_common.h): categories (from #dr_category_t)
+ * to indicate the type of operation performed (e.g., a load, a math operation, floating
+ * point math operation, a branch, etc.), arithmetic flags, number of source and
+ * destination register operands, register operands and their size (which is the same for
+ * all operands), instruction length, and ISA mode (which is DR_ISA_REGDEPS).
+ * Querying additional information (e.g., the instruction opcode) will return the default
+ * value generated by instr_create() (i.e., zero).
+ */
+
+/* XXX i#1684: Note that this is part of a larger issue, where lack of cross-arch support
+ * in the same build of DynamoRIO is limiting us.
```

Comment:
I don't understand this comment being here

Reply:
This is stale. Removed.